### PR TITLE
Integration test for boot-inspect

### PR DESCRIPTION
**Pull request by EricEdens**
This adds an integration test for testing boot disk inspection. It currently has three cases (more to come in a future PR); here's what each shows:
 * SLES SAP: Failing test -- which is good! (Detection thinks this is SLES, not SLES  for SAP)
 * OpenSUSE 15: libguestfs on Debian 9 doesn't detect OpenSUSE 15, so this shows that the library *is* able to detect it.
 * Windows 32-bit: Shows that we can detect 32-bit

PR to add  to prow: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/509

## Testing
* Ran locally using steps described from README
* Ran on Prow using pj-on-kind
    * spyglass view of test results: https://oss-prow.knative.dev/view/gs/edens-test-prow/e2e/1305548384181948416/

### CODE REVIEW

File: cli_tools_integ_tests/boot_inspect/boot_inspect_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
+package boot_inspect
```
Comment (zoran15) [inline cli_tools_integ_tests/boot_inspect/boot_inspect_test.go:1]: copyright?

File: cli_tools_integ_tests/boot_inspect/boot_inspect_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+			disk.InspectionResult{
+				Architecture: "x86",
+				Distro:       "windows",
+				Major:        "6",
+				Minor:        "1",
+			},
+		},
+	} {
+		// Without this, each parallel test will reference the same tc instance.
+		tt := tt
```
Comment (zoran15) [inline cli_tools_integ_tests/boot_inspect/boot_inspect_test.go:70]: how does this work?

**Review by zoran15 — COMMENTED** [review]
These are not integration tests, since there's no cross-module integration being tested here. It's more of a full unit test for boot-inspect. Or, if you treat boot-inspect as a separate module, it's an e2e test. I don't think cli_tools_integ_tests is an appropriate folder for it. It should either go to cli_tools_e2e_test or next to the prod code in cli_tools/common/disk.

File: cli_tools_integ_tests/boot_inspect/boot_inspect_test.go (hunk 1 above)
Comment (EricEdens) [reply cli_tools_integ_tests/boot_inspect/boot_inspect_test.go]: fixed, thanks!

File: cli_tools_integ_tests/boot_inspect/boot_inspect_test.go (hunk 2 above)
Comment (EricEdens) [reply cli_tools_integ_tests/boot_inspect/boot_inspect_test.go]: Good question .. added a reference to the golang FAQ (https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables)

**Comment by EricEdens** [comment]
> It should either go to cli_tools_e2e_test or next to the prod code in cli_tools/common/disk.

SGTM. Moved to the e2e directory.

**Review by zoran15 — APPROVED** [review]

File: cli_tools_e2e_test/diskinspect/disk_inspect_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+				Architecture: "x86",
+				Distro:       "windows",
+				Major:        "6",
+				Minor:        "1",
+			},
+		},
+	} {
+		// Without this, each parallel test will reference the last tt instance.
+		// https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables
+		tt := tt
```
Comment (dntczdx) [inline cli_tools_e2e_test/diskinspect/disk_inspect_test.go:85]: minor: even I know we need a local var for this purpose, I am still a bit confused here. A different var name (currentTest?) can be better.

**Review by dntczdx — APPROVED** [review]

File: cli_tools_e2e_test/diskinspect/disk_inspect_test.go (hunk 3 above)
Comment (EricEdens) [reply cli_tools_e2e_test/diskinspect/disk_inspect_test.go]: done!

**Pull request merged**